File: fsharp/dnd-character/DndCharacter.fs
module DndCharacter

open System

let rand = Random()
let rolld6 = fun f -> rand.Next(1,6)

type Character = {
Strength : int;
Dexterity: int;
Constitution : int;
Intelligence : int;
Wisdom : int;
Charisma : int;
Hitpoints : int;
};

let modifier x = Math.Floor(((float x - 10.0) / 2.0)) |> int

let ability() = 
    [|0..4|]
    |> Array.map rolld6
    |> Array.sortByDescending id
    |> Array.take 3
    |> Array.sum


let createCharacter() =
    let con = ability()
    let hp = 10 + (modifier con)
    {Strength= ability();Dexterity = ability(); Constitution= con;Intelligence = ability(); Wisdom = ability(); Charisma = ability(); Hitpoints = hp}

File: java/dnd-character/src/main/java/DnDCharacter.java
import java.util.Random;

class DnDCharacter {
    private static final Random _rand = new Random();
    private final Integer _str = ability();
    private final Integer _dex = ability();
    private final Integer _con = ability();
    private final Integer _wis = ability();
    private final Integer _int = ability();
    private final Integer _cha = ability();


    int ability() {
        return _rand.ints(4,1,7).sorted().skip(1).sum();
    }

    int modifier(int input) {
        return (int) Math.floor((input - 10) / 2.0);
    }

    int getStrength() {
        return _str;
    }

    int getDexterity() {
        return _dex;
    }

    int getConstitution() {
        return _con;
    }

    int getIntelligence() {
        return _int;
    }

    int getWisdom() {
        return _wis;
    }

    int getCharisma() {
        return _cha;
    }

    int getHitpoints() {
        return modifier(_con) + 10;
    }

}
